### PR TITLE
Support multiple users with the same name for different host for 1st gen.

### LINKED ISSUE #1057: Postgres user creation again and again
### Terraform Version
Terraform v0.11.3
+ provider.google v1.5.0

### Affected Resource(s)
- google_sql_user

### Terraform Configuration Files
```hcl
resource "google_sql_user" "root" {
  name     = "${var.db_root_username}"
  instance = "${google_sql_database_instance.master.name}"
  host     = "no-host"
  password = "${var.db_root_password}"
}

resource "google_sql_user" "user" {
  name     = "${var.db_user_username}"
  instance = "${google_sql_database_instance.master.name}"
  host     = "host"
  password = "${var.db_user_password}"
}
```
### Expected Behavior
The creation of users the first time, and nothing after.

### Actual Behavior
The users are destroyed and created again each time I use Terraform.

```
google_sql_user.user: Refreshing state... (ID: pg-1/booking)
google_sql_user.root: Refreshing state... (ID: pg-1/postgres)
...
-/+ google_sql_user.root (new resource required)
      id:       "pg-1/postgres" => <computed> (forces new resource)
      host:     "" => "no-host" (forces new resource)
      instance: "pg-1" => "pg-1"
      name:     "postgres" => "postgres"
      password: <sensitive> => <sensitive> (attribute changed)
      project:  "sr-services-core1-prod-0725" => <computed>

-/+ google_sql_user.user (new resource required)
      id:       "pg-1/booking" => <computed> (forces new resource)
      host:     "" => "host" (forces new resource)
      instance: "pg-1" => "pg-1"
      name:     "booking" => "booking"
      password: <sensitive> => <sensitive> (attribute changed)
      project:  "sr-services-core1-prod-0725" => <computed>

Plan: 2 to add, 0 to change, 2 to destroy.
```

### Steps to Reproduce
1. `terraform apply`

### PR DESCRIPTION
**Pull request by rosbo**
Fixes #1057 

* support multiple users with the same name for different host for v1 instances.
* make the host field optional.
* Added an acceptance test for 2nd gen sql user
* Merge import test into regular tests.
* Improved documentation for sql user
    * Mention that host is not supported for 2nd gen sql instance.
* Improved documentation for sql instance
    * The example was wrong. Tier `D0` for a `POSTGRES_9_6` instance is not a valid tier.
    *  Added an additional example. One for 1st gen and one for 2nd gen instances.

### CODE REVIEW

**Comment by rosbo** [comment]
```sh
--- PASS: TestAccGoogleSqlUser_firstGen (86.56s)
--- PASS: TestAccGoogleSqlUser_secondGen (508.87s)
```

File: google/resource_sql_user_test.go [hunk 1]
```diff
@@ -107,39 +125,48 @@ func testAccGoogleSqlUserDestroy(s *terraform.State) error {
...
 		region = "us-central"
 		settings {
 			tier = "D0"
 		}
 	}
 
-	resource "google_sql_user" "user" {
-		name = "user%s"
+	resource "google_sql_user" "user1" {
+		name = "john"
```
Comment (danawillow) [inline google/resource_sql_user_test.go]: I'm sorry to be That Person but can we use a gender-neutral name here?

File: website/docs/r/sql_database_instance.html.markdown [hunk 2]
```diff
@@ -19,19 +19,39 @@ a restricted host and strong password.
...
-Example creating a SQL Database.
+### SQL First Generation
 
 ```hcl
 resource "google_sql_database_instance" "master" {
   name = "master-instance"
-  database_version = "POSTGRES_9_6"
+  database_version = "MYSQL_5_6"
+  # First-generation instance regions are not the conventional
+  # Google Compute Engine region. See argument reference below.
```
Comment (danawillow) [inline website/docs/r/sql_database_instance.html.markdown]: nit: regions

File: website/docs/r/sql_database_instance.html.markdown [hunk 3]
```diff
@@ -19,19 +19,39 @@ a restricted host and strong password.
...
+### SQL Second generation
+
+```hcl
+resource "google_sql_database_instance" "master" {
+  name = "master-instance"
+  database_version = "POSTGRES_9_6"
+  region = "us-central1"
+
+  settings {
+    # Second-generation instance tier are based on the machine
```
Comment (danawillow) [inline website/docs/r/sql_database_instance.html.markdown]: nit: tiers

File: website/docs/r/sql_user.html.markdown [hunk 4]
```diff
@@ -52,6 +49,10 @@ The following arguments are supported:
 
 - - -
 
+* `host` - (Optional) The host the user can connect from. This is only supported
+    for first generation SQL instance. Don't set this field for second generation
```
Comment (danawillow) [inline website/docs/r/sql_user.html.markdown]: nit: instances

File: website/docs/r/sql_user.html.markdown [hunk 5]
```diff
@@ -52,6 +49,10 @@ The following arguments are supported:
...
+    SQL instance. Can be an IP address. Changing this forces a new resource to be created.
```
Comment (danawillow) [inline website/docs/r/sql_user.html.markdown]: nit: instances

File: google/resource_sql_user_test.go (hunk 1 above)
Comment (rosbo) [reply google/resource_sql_user_test.go]: Sure. Changed for `admin`.

File: website/docs/r/sql_database_instance.html.markdown (hunk 2 above)
Comment (rosbo) [reply website/docs/r/sql_database_instance.html.markdown]: Done

File: website/docs/r/sql_database_instance.html.markdown (hunk 3 above)
Comment (rosbo) [reply website/docs/r/sql_database_instance.html.markdown]: Done

File: website/docs/r/sql_user.html.markdown (hunk 4 above)
Comment (rosbo) [reply website/docs/r/sql_user.html.markdown]: done

File: website/docs/r/sql_user.html.markdown (hunk 5 above)
Comment (rosbo) [reply website/docs/r/sql_user.html.markdown]: done

**Pull request merged**